Assets/MixedRealityToolkit/_Core/Utilities/ClipPlane.cs

```diff
@@ -62,95 +76,114 @@ private void OnDrawGizmosSelected()
 
         private void OnDestroy()
         {
-            if (materials != null)
+            if (Renderers == null)
             {
-                foreach (Material material in materials)
-                {
-                    if (Application.isPlaying)
-                    {
-                        Destroy(material);
-                    }
-                }
-
-                materials = null;
+                return;
             }
-        }
-
-        private void Initialize()
-        {
-            clipPlaneId = Shader.PropertyToID("_ClipPlane");
-
-            materials = new Material[renderers.Length];
 
-            for (int i = 0; i < renderers.Length; ++i)
+            foreach (Renderer renderer in Renderers)
```

Comment:
for loop change?

Reply:
we have standardized on for vs foreach. consistency is good :)

Reply:
I'm making the requested changes now, I'm confused on what to change here?

Reply:
let's revert it back to the for loop for perf

Reply:
Ah okay, I'll make that change to conform to the standard. Is the perf worry about allocations? If so, I believe Unity fixed that in 5.6: https://forum.unity.com/threads/upgraded-c-compiler-on-5-3-5p8.417363/ "this generally means that doing 'foreach' over an Array, List<T>, Dictionary<T>, or HashSet<T> should not allocate."

Reply:
Yes mainly for allocations, but even on platforms like HoloLens, the perf difference is pretty noticeable in certain hot code areas.

Reply:
Sounds good!